Ginger/Ginger/ApplicationModelsLib/POMModels/POMWizardLib/LearnWizard/AIGeneratedPreviewWizardPage.xaml.cs

```diff
@@ -0,0 +1,242 @@
+                string directory = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
+
+                // Combine the directory with the file name
+                string filePath = Path.Combine(directory, "OpenAIappsetting.json");
+
+                string json = File.ReadAllText(filePath);
+
+
+                var root = JsonConvert.DeserializeObject<Root>(json);
+
+                return root.ApiSettings;
+
+            }
+
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to load OpenAI API settings", ex);
+                return null;
+            }
+        }
+
+        private async void GenerateHtmlAsync()
+        {
+            if(!string.IsNullOrEmpty(mWizard.HtmlFilePath))
+            {
+                xGenerateAIPanel.Visibility = Visibility.Collapsed;
+                return;
+            }
+            mWizard.ProcessStarted();
+            string fileName = Path.GetFileName(mWizard.ScreenShotImagePath);
+            xGenerateAIPanel.Visibility = Visibility.Visible;
+            await GenerateResponseAsync(fileName: fileName, FilePath: mWizard.ScreenShotImagePath);
+            if (!string.IsNullOrEmpty(mWizard.HtmlFilePath))
+            {
+                MyWebView.Source = new Uri(mWizard.HtmlFilePath);
+                xGenerateAIPanel.Visibility = Visibility.Collapsed;
+                MyWebView.Visibility = Visibility.Visible;
+                xViewGenerateHTMLButton.Visibility = Visibility.Visible;
+            }
+            else
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to load OpenAI API Response");
+                xGenerateAIPanel.Visibility = Visibility.Collapsed;
+                MyWebView.Visibility = Visibility.Collapsed;
+            }
+            xReGenerateButton.Visibility = Visibility.Visible;
+            mWizard.ProcessEnded();
+        }
+
+        private async Task GenerateResponseAsync(string fileName, string FilePath = null)
+        {
+            // Extract filename with extension
+            string fileNameWithExtension = Path.GetFileName(FilePath);
+
+            // Remove the extension
+            string fileNameWithoutExtension = Path.GetFileNameWithoutExtension(fileNameWithExtension);
+
+            string response = await GetChatGPTResponse(FilePath);
+            if(!response.EndsWith("```"))
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Re-generate the Preview or faced same issue again contact to system administration");
+                return;
+            }
+            response = response.Replace("```html", "").Replace("```", "");
+
+            var htmlDoc = new HtmlDocument();
+            htmlDoc.LoadHtml(response);
+            string SolutionFolder = WorkSpace.Instance.Solution.Folder;
+            string targetPath = System.IO.Path.Combine(SolutionFolder, @"Documents\AIGeneratePreview");
```

Comment:
Use Path.DirectorySeparatorChar instead of \

Reply:
done